### ISSUE TITLE
What is the design of multiple nodes?

### ISSUE THREAD

**Issue by tisonkun** [engula/microunit#4]
When working on #2, it confuses me a bit that we're now always return one node on `list_nodes`.

As an orchestration framework we must have a design on coordinate multiple nodes. Do we have a design talk about it?

---

Besides, for this kind of open-ended discussion, you may think of turning on [GitHub Discussion](https://docs.github.com/en/discussions) for distinguishing them from work item and bug tracking in issues. You can omit this message if it's not clear to you - I'm going to write down a post talking about how this feature does goods for your community.

**Comment by tisonkun** [comment]
[The Engula Design](https://github.com/engula/engula/blob/main/docs/design.md) provides some information. Now I know how the concept "node" and "unit" stands and their mapping. However, `list_nodes` still confuses me - either we create a node coordinator, or we talk with the cluster manager (k8s apiserver, for example) to get all nodes in a cluster.

**Comment by huachaohuang** [comment]
Yep, sorry for that. The whole architecture is still under design.

**Comment by tisonkun** [comment]
> Yep

The question is an open-ended question though. I don't know what is yes 😄 

> Sorry for that. The whole architecture is still under design.

Never mind. I create this issue to make these points more clear - it is how the community creates value.
